src/main/java/net/sf/jabref/logic/exporter/CustomExportList.java

```diff
@@ -93,20 +99,22 @@ private void readPrefs(JabRefPreferences prefs) {
         } else {
             lfFileName = s.get(1);
         }
-        ExportFormat format = new ExportFormat(s.get(0), s.get(0), lfFileName, null, s.get(2));
+        ExportFormat format = new ExportFormat(s.get(0), s.get(0), lfFileName, null, s.get(2), layoutPreferences,
+                savePreferences);
         format.setCustomExport(true);
         return Optional.of(format);
     }
 
-    public void addFormat(List<String> s) {
-        createFormat(s).ifPresent(format -> {
+    public void addFormat(List<String> s, LayoutFormatterPreferences layoutPreferences, SavePreferences savePreferences) {
+        createFormat(s, layoutPreferences, savePreferences).ifPresent(format -> {
             formats.put(format.getConsoleName(), format);
             list.add(s);
         });
     }
 
-    public void remove(List<String> toRemove) {
-        createFormat(toRemove).ifPresent(format -> {
+    public void remove(List<String> toRemove, LayoutFormatterPreferences layoutPreferences,
```

Comment:
Is it really necessry to pass layoutPreferences and savePreferences? Might there be two custom exporters with the same names, but different preferences? Does that make sense?